src/main/java/org/junit/Assert.java

```diff
@@ -1010,16 +1015,22 @@ public static void assertThrows(Class<? extends Throwable> expectedThrowable, Th
                     expected += "@" + Integer.toHexString(System.identityHashCode(expectedThrowable));
                     actual += "@" + Integer.toHexString(System.identityHashCode(actualThrowable));
                 }
-                String mismatchMessage = format("unexpected exception type thrown;", expected, actual);
+                String mismatchMessage = buildPrefix(message)
+                        + format("unexpected exception type thrown;", expected, actual);
 
                 // The AssertionError(String, Throwable) ctor is only available on JDK7.
                 AssertionError assertionError = new AssertionError(mismatchMessage);
                 assertionError.initCause(actualThrown);
                 throw assertionError;
             }
         }
-        String message = String.format("expected %s to be thrown, but nothing was thrown",
-                formatClass(expectedThrowable));
-        throw new AssertionError(message);
+        String notThrownMessage = buildPrefix(message) + String
+                .format("expected %s to be thrown, but nothing was thrown",
+                        formatClass(expectedThrowable));
+        throw new AssertionError(notThrownMessage);
+    }
+
+    private static String buildPrefix(String message) {
+        return message != null && message.length() != 0 ? message + " ==> " : "";
```

Comment:
Can you please replace `==>` with something unobstrusive like a colon `:`. Thanks.